### DISCUSSION TITLE
vertico-buffer-display-action reuse-some-window doesn't work if emacs windows are very large

### DISCUSSION THREAD

**Discussion by petergaultney** [minad/vertico#527] [Q&A]
<img width="991" alt="Screenshot 2024-09-25 at 12 41 21" src="https://github.com/user-attachments/assets/0a072ba2-62ab-4c2a-acef-95dc6762b3e1">

I've used `customize` to set the value to `display-buffer-reuse-window`. 

I can confirm that `vertico-buffer-mode` is turned on (and that turning it off does actually turn it off, and vice-versa).

But for whatever reason, whenever i open up a vertico  buffer (e.g. with find-file), it opens in a new window that is approximately half the size of the current window, oriented vertically (within the existing window). Which is especially weird because then find-file itself will open the file in that newly-created window, which I didn't want.

![Screenshot 2024-09-25 at 12 49 06](https://github.com/user-attachments/assets/13936f38-3ea1-4019-be87-245fc886d44f)

I'm using elpaca to install vertico, and the installed version that I have is from August 22nd, and says its the version 1.9 tag.

I sort of doubt this is a bug in vertico exactly, mainly because no-one else has reported it... but I'm at a loss as to how to debug why vertico is doing what it's doing.

**Comment by petergaultney** [comment]
wait, no, i have _some_ idea - vertico seems to be reacting to the size of the windows/frames? And if my font is smaller (and the windows themselves are therefore "larger"), then suddenly it switches its behavior. Is there any way to disable this? I'd really like to have it actually reuse an existing window rather than create a new one, regardless of how many windows I have open.

**Comment by petergaultney** [comment]
`least-recently-used-window` seems to work 'correctly' no matter the window size.

**Comment by minad** [comment]
As far as I understand `display-buffer-reuse-window` tries to reuse a window which already displays the given buffer. In the context of Vertico it is a bit questionable, since the minibuffer is not already shown in a regular window. I've chosen it as the default back then since it seems to the right thing in most scenarios, but I don't think there are reliable guarantees. `display-buffer-use-least-recent-window` was added in 28.1.

EDIT: I've pushed https://github.com/minad/vertico/commit/9958f1ff4a2828623c48ec3a2fa4b516964a90b3 changing the default display action. Let's see if we get user feedback about this change.
